BibBuddy/app/src/main/java/de/bibbuddy/BookBarcodeScannerFragment.java

```diff
@@ -116,37 +116,83 @@ public void release() {
       }
 
       /**
-       * Receives the ISBN of a book.
+       * Receives a barcode ISBN, hands the ISBN over the the API.
        */
       @Override
       public void receiveDetections(Detector.Detections<Barcode> detections) {
         final SparseArray<Barcode> barcodes = detections.getDetectedItems();
         if (barcodes.size() != 0) {
+          barcodeDetector.release();
+          handleIsbnInput(barcodes.valueAt(0).displayValue);
+        }
+      }
+    });
+  }
 
-          barcodeText.post(new Runnable() {
-
-            @Override
-            public void run() {
-              barcodeData = barcodes.valueAt(0).displayValue;
-              String barcodeDisplay = getString(R.string.barcode_value) + " " + barcodeData;
-              barcodeText.setText(barcodeDisplay);
-              // TODO get display message with ISBN successfully scanned
-              // TODO temporary stop the barcode scanner
-              // TODO new fragment user wait for search
-              // TODO check if book exists in API
-              // TODO API get book data
-              // if not exists: display error message and open create new book form alternative
-              // TODO check if valid data / book information data display
-              // TODO add book to shelf
-              // TODO bundle for BookFragment with shelfId and shelfName
-            }
-          });
-
+  private void handleIsbnInput(String isbn) {
+    isbnRetriever = new IsbnRetriever(isbn);
+    System.out.println(isbnRetriever);
+    thread = new Thread(isbnRetriever);
+    thread.start();
+
+    try {
+      thread.join();
+    } catch (Exception e) {
+      System.out.println(e);
+    }
+
+    // retrieve metadata that was saved
+    Book book = isbnRetriever.getBook();
+    closeFragment();
+    if (book != null) {
+      handleAddBook(book);
+    } else {
+      
+      getActivity().runOnUiThread(new Runnable() {
+        public void run() {
+          Toast.makeText(getActivity(), getString(R.string.isbn_not_found),
+              Toast.LENGTH_SHORT).show();
         }
+      });
+    }
+  }
+
+  private void handleAddBook(Book book) {
+    // TODO authors of a book
+    BookDao bookDao = new BookDao(new DatabaseHelper(getContext()));
+    bookDao.create(book, new ArrayList<Author>(), shelfId);
+
+    getActivity().runOnUiThread(new Runnable() {
+      public void run() {
+        Toast.makeText(getActivity(), getString(R.string.added_book),
+            Toast.LENGTH_SHORT).show();
       }
     });
+
+    closeFragment();
+  }
+
+  /**
+   * Closes the BookBarcodeScannerFragment.
+   */
+  public void closeFragment() {
+    LibraryFragment fragment = new LibraryFragment();
```

Comment:
Would it be better to open the BookFragment where you see that the new book was added?